common/smoke-test/Initialize-SmokeTests.ps1

```diff
@@ -90,20 +90,28 @@ function Set-EnvironmentVariables {
 
 function New-DeployManifest {
   Write-Verbose "Detecting samples..."
-  $javascriptSamples = (Get-ChildItem -Path "$repoRoot/sdk/$ServiceDirectory/*/samples/javascript/" -Directory
-    | Where-Object { Test-Path "$_/package.json" })
+  $packageDir = Get-ChildItem -Directory -Path "$repoRoot/sdk/$ServiceDirectory/*"
+
+  $javascriptSamples = $packageDir | ForEach-Object {
+    $versions = (Get-Item "$_/samples/*").Name
+    $newestVer = $versions | Sort-Object {[int]($_ -replace '^v' -replace '(\d+).*', '$1')} -Descending | Select-Object -First 1
```

Comment:
Are these guaranteed to always be single integer version numbers?

Reply:
I think that's the plan, but there might be "beta" versions, I don't think there will be any multi-integer versions though.

Reply:
The other alternative is to update our semver class with format validation support for `vN` format and/or allow it to ignore validation.

Reply:
After syncing with Will, the only version formats will be v{currentMajor} and v{currentMajor}-beta

Reply:
If there is a "v2" and "v2-beta" which should sort higher?

Reply:
-beta